app/src/main/java/is/xyz/mpv/preferences/PreferenceActivity.kt

```diff
@@ -0,0 +1,167 @@
+        }
+
+        if (savedInstanceState != null) {
+            title = savedInstanceState.getCharSequence("title")
+        } else {
+            supportFragmentManager.beginTransaction().replace(R.id.main, SettingsFragment())
+                .commit()
+            supportFragmentManager.addOnBackStackChangedListener {
+                if (supportFragmentManager.backStackEntryCount == 0) {
+                    setTitle(R.string.title_activity_settings)
+                }
+            }
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putCharSequence("title", title)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        preferences.unregisterOnSharedPreferenceChangeListener(this)
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String?) {
+        if (key != "material_you_theming") return
+        if (sharedPreferences.getBoolean(key, false)) {
+            DynamicColors.applyToActivityIfAvailable(this)
+        }
+        recreate()
+    }
+
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressedDispatcher.onBackPressed()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+
+    override fun onPreferenceStartFragment(
+        caller: PreferenceFragmentCompat, pref: Preference
+    ): Boolean {
+        val fragment = supportFragmentManager.fragmentFactory.instantiate(
+            classLoader, pref.fragment ?: return false
+        ).apply { arguments = pref.extras }
+
+        supportFragmentManager.beginTransaction().replace(R.id.main, fragment).addToBackStack(null)
+            .commit()
+
+        title = pref.title
+        return true
+    }
+
+    /**
+     * The root preference fragment that displays preferences that link to the other preference
+     * fragments below.
+     */
+    class SettingsFragment : PreferenceFragmentCompat() {
+        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+            setPreferencesFromResource(R.xml.preferences_root, rootKey)
+        }
+    }
+
+
+    class GeneralPreference : PreferenceFragmentCompat(), Preference.OnPreferenceChangeListener {
+        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+            setPreferencesFromResource(R.xml.pref_general, rootKey)
+            // Hide Material You on Android 11 or lower
+            preferenceManager.findPreference<SwitchPreferenceCompat>("material_you_theming")?.isVisible =
+                (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S)
+        }
+
+        override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+            if (newValue == true) DynamicColors.applyToActivityIfAvailable(requireActivity())
```

Comment:
missing key check
but also isn't this covered by `onSharedPreferenceChanged` in the activity itself?